src/index.js

```diff
@@ -111,35 +135,139 @@ IdTokenVerifier.prototype.verify = function(token, nonce, cb) {
     if (err) {
       return cb(err);
     }
+
     if (rsaVerifier.verify(headAndPayload, signature)) {
```

Comment:
Instead of wrapping everything in this; check the negated version and fail immediately. Keeps the indentation one level lower.